R/equivalence_test.R

```diff
@@ -48,21 +53,42 @@ equivalence_test.numeric <- function(posterior, bounds = "default", ci = .95, ve
     )
   }
 
+  out$HDI_low <- attr(rope_data, "HDI_area", exact = TRUE)[1]
+  out$HDI_high <- attr(rope_data, "HDI_area", exact = TRUE)[2]
+
+  class(out) <- c("equivalence_test", class(out))
   out
 }
 
 
+#' @export
+print.equivalence_test <- function(x, ...) {
+  cat("# Test for Practical Equivalence\n\n")
+  cat(sprintf("  ROPE: [%.2f %.2f]\n\n", x$ROPE_low[1], x$ROPE_high[1]))
+  ci <- x$CI[1]
+
+  x$ROPE_Percentage <- sprintf("%.2f%%", x$ROPE_Percentage)
+  x$HDI <- sprintf("[%.2f %.2f]", x$HDI_low, x$HDI_high)
+
+  x <- x[, intersect(c("Parameter", "ROPE_Equivalence", "ROPE_Percentage", "HDI"), colnames(x))]
+
+  colnames(x)[which(colnames(x) == "ROPE_Equivalence")] <- "H0"
+  colnames(x)[which(colnames(x) == "ROPE_Percentage")] <- "Inside ROPE"
```

Comment:
It might be clearer adding "Proportion" or "Percentage" or "%", e.g; "% inside ROPE"